### PR TITLE
Exam Questions Trish DeLozier

### PR DESCRIPTION
**Pull request by TrishDeLozier**

### CODE REVIEW

**Comment by BillWagner** [comment]
@TrishDeLozier I don't see an updated doc file with your answers. Is that somewhere else? I'd like to look at it to get a better idea where you are.

File: QueryExpressionPattern/Program.cs [hunk 1]
```diff
@@ -38,33 +38,60 @@ static void Main(string[] args)
...
-            var smallNumbers = numbers.Where((n) => n < 5);
-
-            var allNumbers = numbers.Select(n => n);
-
-            var squares = numbers.Select(n => new { Number = n, Square = n * n });
-
-            var people = employees.Where(e => e.Age > 30).
+            
+            var smallNumbers = //numbers.Where((n) => n < 5);
+/*---------->*/from n in numbers
```
Comment (BillWagner) [inline QueryExpressionPattern/Program.cs:43]: Correct.

File: QueryExpressionPattern/Program.cs [hunk 2]
```diff
@@ -38,33 +38,60 @@ static void Main(string[] args)
...
+               where n < 5
+               select n;
+
+            var allNumbers = //numbers.Select(n => n);
+/*---------->*/from n in numbers
```
Comment (BillWagner) [inline QueryExpressionPattern/Program.cs:48]: Correct.

File: QueryExpressionPattern/Program.cs [hunk 3]
```diff
@@ -38,33 +38,60 @@ static void Main(string[] args)
...
+               select n;
+
+            var squares = //numbers.Select(n => new { Number = n, Square = n * n });
+/*---------->*/from n in numbers
```
Comment (BillWagner) [inline QueryExpressionPattern/Program.cs:52]: Correct.

File: QueryExpressionPattern/Program.cs [hunk 4]
```diff
@@ -38,33 +38,60 @@ static void Main(string[] args)
...
                 OrderBy(e => e.LastName).
                 ThenBy(e => e.FirstName).
-                ThenBy(e => e.Age);
-
-            var results = employees.GroupBy(e => e.Department).
-                Select(d => new { Department = d.Key, Size = d.Count() });
-
-            var results2 = employees.GroupBy(e => e.Department).
+                ThenBy(e => e.Age);*/
+/*---------->*/from e in employees
```
Comment (BillWagner) [inline QueryExpressionPattern/Program.cs:60]: Correct.

File: QueryExpressionPattern/Program.cs [hunk 5]
```diff
@@ -38,33 +38,60 @@ static void Main(string[] args)
...
+              where e.Age > 30 
+              orderby e.LastName, e.FirstName, e.Age
+              select e;
+
+            var results =
+                /*employees.GroupBy(e => e.Department).
+                Select(d => new { Department = d.Key, Size = d.Count() });*/
+/*---------->*/from e in employees
```
Comment (BillWagner) [inline QueryExpressionPattern/Program.cs:68]: Perfect.

File: QueryExpressionPattern/Program.cs [hunk 6]
```diff
@@ -38,33 +38,60 @@ static void Main(string[] args)
...
+ 
+            var results2 = 
+                /*employees.GroupBy(e => e.Department).
                 Select(d => new
                 {
                     Department = d.Key,
                     Employees = d.AsEnumerable()
-                });
+                });*/
+/*---------->*/from e in employees
```
Comment (BillWagner) [inline QueryExpressionPattern/Program.cs:80]: Perfect.

File: QueryExpressionPattern/Program.cs [hunk 7]
```diff
@@ -79,20 +106,29 @@ static void Main(string[] args)
...
             var words = new string[] { "one", "two", "three" };
             var romanNumerals = new string[] { "I", "II", "III" };
-            var triples = nums.SelectMany(n => words,
+            var triples =
+                /*nums.SelectMany(n => words,
                 (n, s) => new { n, s }).
                 SelectMany(pair => romanNumerals,
-                (pair, n) => new { Arabic = pair.n, Word = pair.s, Roman = n });
+                (pair, n) => new { Arabic = pair.n, Word = pair.s, Roman = n });*/
+/*---------->*/from n in new int[] { 1, 2, 3 }
```
Comment (BillWagner) [inline QueryExpressionPattern/Program.cs:114]: Perfect.

File: QueryExpressionPattern/Program.cs [hunk 8]
```diff
@@ -79,20 +106,29 @@ static void Main(string[] args)
...
             var digits = new int[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
             var labels = new string[] { "0", "1", "2", "3", "4", "5" };
-            var query = digits.Join(labels, num => num.ToString(), label => label,
-                (num, label) => new { num, label });
-
-            var groups = departments.GroupJoin(employees,
+            var query = 
+                /*digits.Join(labels, num => num.ToString(), label => label,
+                (num, label) => new { num, label });*/
+                /*---------->*/from num in numbers
```
Comment (BillWagner) [inline QueryExpressionPattern/Program.cs:124]: Well done.

File: ExtensionMethods/Program.cs [hunk 9]
```diff
@@ -5,8 +5,36 @@
 
 namespace ExtensionMethods
 {
-    class Program
+    public static class Extensions
     {
+
+        static void GreaterThan(this Person greater)
```
Comment (BillWagner) [inline ExtensionMethods/Program.cs:11]: You're on the right track here. The method signatures are all correct.

File: Loops/Program.cs [hunk 10]
```diff
@@ -6,7 +6,7 @@
 
 namespace Loops
 {
-    class Program
+    class Program : Student
```
Comment (BillWagner) [inline Loops/Program.cs:9]: Hmm. You made this one quite a bit tougher than it should have been.

**Comment by BillWagner** [comment]
Trish,

Some of the code looks good. I'm not going to comment too much until I see the written answers as well.

I'd like to see you try the other coding problems, both for practice and to see how much you've already learned.

**Comment by TrishDeLozier** [comment]
Bill,
Thanks for the feedback! I spent so much time on the linq queries,
extension methods, and student list that I did not even have a chance to do
my written answers. I do however understand what was required form the
written, I just began on the programs first because I knew they would be
more time consuming. I am however still working on it when I can so maybe I
can show you if I've accomplished anything new the next time we meet.
Thanks,
Trish

**Comment by BillWagner** [comment]
Trish,

It will be simpler if you can just email it to me.  

My schedule has me in New England until the end of the month, then Chicago for a week, and then Wisconsin most of the week after that.  

The next time I'm in Detroit is the last day of class. 

I'd have wanted to spend much more time during the program, but the folks in charge of scheduling (who knew that schedule) did not want to move the dates.

Cheers,
Bill Wagner
http://theBillWagner.com / @billwagner / bill.w.wagner@outlook.com / 734.645.9709
Sent from Surface Pro

From: TrishDeLozier
Sent: ‎Tuesday‎, ‎July‎ ‎22‎, ‎2014 ‎12‎:‎28‎ ‎PM
To: BillWagner/ExperienceITExam
Cc: Bill Wagner

Bill, 
Thanks for the feedback! I spent so much time on the linq queries, 
extension methods, and student list that I did not even have a chance to do 
my written answers. I do however understand what was required form the 
written, I just began on the programs first because I knew they would be 
more time consuming. I am however still working on it when I can so maybe I 
can show you if I've accomplished anything new the next time we meet. 
Thanks, 
Trish

**Pull request closed without merging**